lib/compiler.js

```diff
@@ -128,3 +120,34 @@ exports.runWebpackCompiler = function runWebpackCompiler(config, $mobileHelper,
         });
     }
 }
+
+function buildEnvCommandLineParams(config, envData, $logger) {
+    const envFlagNames = Object.keys(envData);
+    const snapshotEnvIndex = envFlagNames.indexOf("snapshot");
+    if (snapshotEnvIndex > -1 && !utils.shouldSnapshot(config)) {
+        logSnapshotWarningMessage($logger);
+        envFlagNames.splice(snapshotEnvIndex, 1);
+    }
+
+    return envFlagNames.map(item => `--env.${item}=${envData[item]}`);
+}
+
+function logSnapshotWarningMessage($logger) {
+    if (!hasLoggedSnapshotWarningMessage) {
+        $logger.warn("Stripping the snapshot flag. " +
+            "Bear in mind that snapshot is only available in release builds and " +
+            "is NOT available on Windows systems.");
+
+        hasLoggedSnapshotWarningMessage = true;
+    }
+}
+
+function getProjectDir($projectData) {
+    const projectDir = safeGet($projectData, "projectDir");
```

Comment:
I don't believe this is necessary - CLI's `$projectData` always has `projectDir` property - if it doesn't the initial creation of the `$projectData` class would have failed long before we get here